test/framework/namespace.go

```diff
@@ -39,20 +40,20 @@ func (f *Framework) CreateNamespace(t *testing.T, ctx *TestCtx) string {
 	}
 
 	namespaceFinalizerFn := func() error {
-		return f.DeleteNamespace(name)
+		return f.DeleteNamespace(ctx, name)
 	}
 
-	ctx.AddFinalizerFn(namespaceFinalizerFn)
+	testCtx.AddFinalizerFn(namespaceFinalizerFn)
 
 	return name
 }
 
-func (f *Framework) DeleteNamespace(name string) error {
-	return f.KubeClient.CoreV1().Namespaces().Delete(f.Ctx, name, metav1.DeleteOptions{})
+func (f *Framework) DeleteNamespace(ctx context.Context, name string) error {
+	return f.KubeClient.CoreV1().Namespaces().Delete(ctx, name, metav1.DeleteOptions{})
 }
 
-func (f *Framework) AddLabelsToNamespace(name string, additionalLabels map[string]string) error {
-	ns, err := f.KubeClient.CoreV1().Namespaces().Get(f.Ctx, name, metav1.GetOptions{})
+func (f *Framework) AddLabelsToNamespace(ctx context.Context, name string, additionalLabels map[string]string) error {
```

Comment:
same here

Reply:
> Good! I think we can take it even further: practically we shouldn't any function in `test/e2` or `test/framework` taking a `ctx context.Context` parameter.

mm but looks like then we are replacing context.TODO() which we had before with context.Background() in all client-go calls. If I remember correctly we wanted to pass them to method calls so instead of context.TODO() we can pass context created at one place. 

For methods that doesn't involve testCtx do you think its fine to pass context object like we have now?. I find it more better :)

Reply:
@simonpasquier Please take a look when you get a chance :)